spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchQueryScan.java

```diff
@@ -79,6 +80,17 @@
     this.splitOpenFileCost = Spark3Util.propertyAsLong(options, SparkReadOptions.FILE_OPEN_COST, null);
   }
 
+  @Override
+  protected Schema snapshotSchema() {
```

Comment:
Why is this method here? The scan should be passed a schema because all of the projection happens in the builder.

Reply:
After working through how Spark constructs a `SparkBatchQueryScan`, I realize that you're right and that this method is not needed in `SparkBatchScan`.

Reply:
Removed `snapshotSchema()` from SparkBatchScan and its subclasses.